pkg/distributor/distributor.go

```diff
@@ -1783,7 +1805,11 @@ func (d *Distributor) sendWriteRequestToBackends(ctx context.Context, tenantID s
 	// errors. For this reason, it's important to give precedence to partition errors, otherwise the distributor may return
 	// a 4xx (ingester error) when it should actually be a 5xx (partition error).
 	if partitionsErr != nil {
-		return partitionsErr
+		if d.cfg.IngestStorageConfig.Migration.IgnoreIngestStorageError {
```

Comment:
The config option has been added under the "migration" block. Given we don't want this setting to be used unless during the migration, I would add an extra check to `DistributorSendToIngestersEnabled`:

```suggestion
		if d.cfg.IngestStorageConfig.DistributorSendToIngestersEnabled && d.cfg.IngestStorageConfig.Migration.IgnoreIngestStorageError {
```

Reply:
Added [here](https://github.com/grafana/mimir/pull/11291/commits/c360ec51ce0568ae8ecc74c26fe638c30ddb4ccd#diff-8e51742acf8f88fa2ae01855fb19125feb5c54c41023f1eaa8beded9f1b4bcc3)